src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java

```diff
@@ -82,4 +86,29 @@ public void handleCheckOutEvent(String data, MessageHeaders messageHeaders) {
       }
     }
   }
+
+  @KafkaListener(
+    id = REQUEST_LISTENER_ID,
+    topicPattern = "#{folioKafkaProperties.listener['request'].topicPattern}",
+    concurrency = "#{folioKafkaProperties.listener['request'].concurrency}")
+  public void handleRequestCancelEvent(String data, MessageHeaders messageHeaders) {
+    String tenantId = getHeaderValue(messageHeaders, XOkapiHeaders.TENANT, null).get(0);
+    var eventData = parseEvent(data);
+    if (Objects.nonNull(eventData) && eventData.getType() == EventData.EventType.CANCEL) {
+      String itemID = eventData.getItemId();
+      if (Objects.nonNull(itemID)) {
+        log.info("updateTransactionStatus:: Received cancel event for itemId: {}", itemID);
+        systemUserScopedExecutionService.executeAsyncSystemUserScoped(tenantId, () ->
+          transactionRepository.findTransactionByItemIdAndStatusNotInClosed(UUID.fromString(itemID))
+            .ifPresent(transactionEntity -> {
+              switch (transactionEntity.getRole()) {
```

Comment:
As the implementation is same for all the roles, I think we don't need to use case and default here. we may call like below.
```suggestion
                  systemUserScopedExecutionService.executeAsyncSystemUserScoped(tenantId, () ->
          transactionRepository.findTransactionByItemIdAndStatusNotInClosed(UUID.fromString(itemID))
            .ifPresent(baseLibraryService::cancelTransactionEntity)
        );
```